[rocdecode] rocDecode utility class documentation (#4475)
ROCm/rocm-systems: projects/rocdecode/docs/reference/rocDecode-ffmpeg-decoder.rst

.. meta::
  :description: rocDecode FFMpeg utility decoder
  :keywords: decode, video decoder, video decoding, ffmpeg, rocDecode, utility functions, AMD, ROCm

***********************************************

> The rocDecode FFMpeg decoder utility class

Reviewer: ```suggestion
rocDecode FFMpeg decoder utility class
```

***********************************************

The rocDecode FFMpeg decoder utility class, `FFMpegVideoDecoder <https://github.com/ROCm/rocm-systems/tree/develop/projects/rocdecode/utils/ffmpegvideodecode/ffmpeg_video_dec.h>`_, is used to decode demultiplexed (demuxed) frames on the CPU. 

.. note:: 

  The ``rocdecode-host`` package must be installed to use FFMpegVideoDecoder.

> FFMpegVideoDecoder uses the FFMpeg video decoding backend to decode frame packets that are output from the FFMpeg demultiplexer (demuxer). It inherits from :doc:`the RocVideoDecode utility class <./rocDecode-util-decoder>` and uses many of the same functions. 

Reviewer: ```suggestion
FFMpegVideoDecoder uses the FFMpeg video decoding backend to decode frame packets output by the FFMpeg demultiplexer (demuxer). It inherits from :doc:`the RocVideoDecode utility class <./rocDecode-util-decoder>` and uses many of the same functions. 
```

> The ``FFMpegVideoDecoder()`` constructor takes the same parameters as ``RocVideoDecoder`` with the exception of the first parameter, which is the number of CPU threads rather than the device ID, and the force zero latency parameter, which isn't supported with the FFMpeg decoder.

Reviewer: ```suggestion
The ``FFMpegVideoDecoder()`` constructor takes the same parameters as ``RocVideoDecoder`` except for the first parameter, which is the number of CPU threads rather than the device ID, and the force zero latency parameter, which isn't supported with the FFMpeg decoder.
```

> Six functions are implemented in FFMpegVideoDecoder: ``DecodeFrame()``, ``GetFrame()``, ``GetOutputSurfaceInfo()``, ``SaveFrameToFile()``, ``ReleaseFrame()``, and ``ReconfigureDecoder()``.

Reviewer: bulleted list might be good here
